component/src/main/java/org/ballerinalang/ftp/util/FTPUtil.java

```diff
@@ -45,29 +48,17 @@ public static boolean validProtocol(String url) {
         return url.startsWith("ftp://") || url.startsWith("sftp://") || url.startsWith("ftps://");
     }
 
-    public static String createUrl(String protocol, String host, long port, String username, String passPhrase,
+    public static String createUrl(String protocol, String host, int port, String username, String passPhrase,
             String basePath) {
-        StringBuilder urlBuilder = new StringBuilder();
-        urlBuilder.append(protocol);
-        urlBuilder.append(":");
-        urlBuilder.append("//");
-        if (username != null && !username.isEmpty()) {
-            urlBuilder.append(username);
-            if (passPhrase != null && !passPhrase.isEmpty()) {
-                urlBuilder.append(":");
-                urlBuilder.append(passPhrase);
-            }
-            urlBuilder.append("@");
-        }
-        urlBuilder.append(host);
-        if (port > 0) {
-            urlBuilder.append(":");
-            urlBuilder.append(port);
-        }
-        if (basePath != null) {
-            urlBuilder.append(basePath);
+        String userInfo = username + ":" + passPhrase;
+        URI uri = null;
+        try {
+            uri = new URI(protocol, userInfo, host, port, basePath, null, null);
+        } catch (URISyntaxException e) {
+            throw new BallerinaConnectorException("Error occurred while constructing a URI from the given components: "
```

Comment:
Let's remove the components wording from here. We can say provided URL information invalid or something similar. WDYT?